servant-foreign/src/Servant/Foreign/Internal.hs

```diff
@@ -211,6 +211,21 @@ instance (KnownSymbol sym, HasForeignType lang ftype t, HasForeign lang ftype su
         { _argName = PathSegment str
         , _argType = ftype }
 
+instance (KnownSymbol sym, HasForeignType lang ftype [t], HasForeign lang ftype sublayout)
+  => HasForeign lang ftype (CaptureAll sym t :> sublayout) where
+  type Foreign ftype (CaptureAll sym t :> sublayout) = Foreign ftype sublayout
+
+  foreignFor lang Proxy Proxy req =
+    foreignFor lang Proxy (Proxy :: Proxy sublayout) $
+      req & reqUrl . path <>~ [Segment (Cap arg)]
```

Comment:
I don't _think_ so, but I have to confess I don't know exactly what `<>~` does. 

AFAICT, all this code has to do is generate signatures of the correct type, rather than capturing values. In which case, it's OK as is, because:

 1. `arg` here is different from the `arg` in `Capture` because here the `ftype` is a proxy around `[t]` rather than `t`.
 2. the test in `ForeignSpec.hs` verifies that it generates valid signatures

But I'll happily admit I don't understand this code very well, so I'd be glad to be corrected.